collectorist/api/src/collector.rs

```diff
@@ -0,0 +1,203 @@
+use actix_web::{delete, get, post, web, HttpResponse, Responder};
+use serde::{Deserialize, Serialize};
+
+use crate::SharedState;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct CollectorConfig {
+    url: String,
+}
+
+/// Register a collector
+#[utoipa::path(
+    post,
+    tag = "collectorist",
+    path = "/api/v1/collector",
+    responses(
+        (status = 200, description = "Collector registered"),
+        (status = BAD_REQUEST, description = "Missing valid id"),
+    ),
+)]
+#[post("/collector/{id}")]
+pub(crate) async fn register_collector(
+    state: web::Data<SharedState>,
+    id: web::Path<String>,
+    config: web::Json<CollectorConfig>,
+) -> actix_web::Result<impl Responder> {
+    if state
+        .collectors
+        .write()
+        .await
+        .register((*id).clone(), (*config).clone())
+        .is_ok()
+    {
+        Ok(HttpResponse::Ok().finish())
+    } else {
+        Ok(HttpResponse::InternalServerError().finish())
+    }
+}
+
+/// Register a collector
+#[utoipa::path(
+    get,
+    tag = "collectorist",
+    path = "/api/v1/collector",
+    responses(
+        (status = 200, description = "Collector configuration located"),
+        (status = BAD_REQUEST, description = "Missing valid id"),
+    ),
+)]
+#[get("/collector/{id}")]
+pub(crate) async fn collector_config(
+    state: web::Data<SharedState>,
+    id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    if let Some(config) = state.collectors.read().await.collector_config(id.clone()) {
+        Ok(HttpResponse::Ok().json(config))
+    } else {
+        Ok(HttpResponse::InternalServerError().finish())
+    }
+}
+
+/// Register a collector
+#[utoipa::path(
+    delete,
+    tag = "collectorist",
+    path = "/api/v1/collector",
+    responses(
+        (status = 200, description = "Collector unregistered"),
+        (status = NOT_FOUND, description = "Collector not found"),
+        (status = BAD_REQUEST, description = "Missing valid id"),
+    ),
+)]
+#[delete("/collector/{id}")]
+pub(crate) async fn deregister_collector(
+    state: web::Data<SharedState>,
+    id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    if let Ok(result) = state.collectors.write().await.deregister((*id).clone()) {
+        if result {
+            Ok(HttpResponse::Ok().finish())
+        } else {
+            Ok(HttpResponse::NotFound().finish())
+        }
+    } else {
+        Ok(HttpResponse::InternalServerError().finish())
+    }
```

Comment:
Since you're already returning an actix `Result`, I would prefer using an actix Error, similar to https://github.com/trustification/trustification/blob/39bfd51c81b69f8ccf831dfe654ab389a4e21a5c/bombastic/api/src/server.rs#L46-L81

The nested if's to return various HttpResponse's can get unwieldy. These aren't so bad, so we could defer introducing an Error if you prefer.